### PR TITLE
update: Rename Redis to Aiven for Caching

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
## Describe your changes

In compliance with recent directives from Redis, this PR updates our product names and URLs, replacing **Aiven for Redis** with **Aiven for Caching** and **Caching** for short. 

## Checklist

- [x] The first paragraph of the page is on one line.
- [x] The other lines have a line break at 90 characters.
- [x] I checked the output.
- [x] I applied the [style guide](../styleguide.md).
- [x] My links start with `/docs/`.

### CODE REVIEW

File: docs/products/caching.md [hunk 1]
```diff
@@ -1,11 +1,10 @@
 ---
-title: Aiven for Redis®*
+title: Aiven for Caching
 ---
 
-Aiven for Redis®\* is a fully managed **in-memory NoSQL database**,
+Aiven for Caching is a fully managed **in-memory NoSQL database**,
```
Comment (hoang-vo) [inline docs/products/caching.md:5]: can we add a bit about the name change by link to our blog post https://aiven.io/blog/aiven-for-redis-becomes-aiven-for-caching

File: docs/products/caching/concepts/lua-scripts-caching.md [hunk 2]
```diff
@@ -1,15 +1,14 @@
 ---
-title: Lua scripts with Aiven for Redis®*
+title: Lua scripts with Aiven for Caching
 ---
 
-Learn how to leverage the inbuilt support for Lua scripting in Redis®\*
-using Aiven.
+Learn how to leverage the inbuilt support for Lua scripting in Redis®* using Aiven for Caching.
```
Comment (hoang-vo) [inline docs/products/caching/concepts/lua-scripts-caching.md:5]: we probably wanna chagen this sentence, something like in Aiven for Caching?

File: docs/products/caching/concepts/lua-scripts-caching.md [hunk 3]
```diff
@@ -1,15 +1,14 @@
...
 
-Redis®\* has inbuilt support for running Lua scripts to perform various
+Redis has inbuilt support for running Lua scripts to perform various
```
Comment (hoang-vo) [inline docs/products/caching/concepts/lua-scripts-caching.md:7]: same here

File: docs/products/caching/concepts/memory-usage.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
+---
+title: Memory Management and persistence in Aiven for Caching
+---
+
+Learn how Aiven for Caching, compatible with legacy Redis® OSS, addresses the challenges of high memory usage and high change rate. Discover how it implements robust memory management and persistence strategies.
```
Comment (hoang-vo) [inline docs/products/caching/concepts/memory-usage.md:5]: i think we should add version 7.2.4

File: docs/products/caching/concepts/memory-usage.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+Aiven for Caching, compatible with legacy Redis® OSS, primarily functions as a
```
Comment (hoang-vo) [inline docs/products/caching/concepts/memory-usage.md:7]: here too

File: docs/products/caching/concepts/overview.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
+---
+title: Aiven for Caching overview
+---
+
+Aiven for Caching, a fully managed **in-memory NoSQL database**, offers efficient storage and quick data access in your preferred cloud. Compatible with legacy Redis® OSS, this service facilitates seamless transitions and compatibility.
```
Comment (hoang-vo) [inline docs/products/caching/concepts/overview.md:5]: and here

File: docs/products/caching/concepts/overview.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+geospatial indexes, and streams.
+
+## Features and benefits of Aiven for Caching
+
+Aiven for Caching has many features that make it easy and stress-free to
+use:
+
+-   **Managed service:** Fully managed service, so you don't
+    have to worry about setup, management, or updates. Aiven provides
+    tools and integrations to help you use Redis in your data
```
Comment (hoang-vo) [inline docs/products/caching/concepts/overview.md:23]: maybe we need to change the sentence to not include Redis?

File: docs/products/caching/concepts/overview.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    network connectivity using VPC peering, PrivateLink, or
+    TransitGateway technologies. Aiven integrates with various
+    observability tooling, including Datadog, Prometheus, and Jolokia,
+    or you can use Aiven's observability tools for improved monitoring
+    and logging.
+-   **DevOps-friendly management and development:** Manage your caching solution
+    using [Aiven Console](https://console.aiven.io/),
+    [Aiven CLI](https://github.com/aiven/aiven-client), or
+    [Terraform](/docs/tools/terraform) .
+    Features like scaling, forking, and upgrading your Redis-based cluster
```
Comment (hoang-vo) [inline docs/products/caching/concepts/overview.md:39]: Aiven for caching cluster

File: docs/products/caching/concepts/overview.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    are simple and efficient. Compatible with open-source software, it integrates
+    with your existing applications and facilitates cloud and regional migrations.
+-   **Backups and disaster recovery:** Automatic and configurable backups ensure data
+    safety. Backups are performed every 24 hours, with retention periods varying by
+    service plan.
+
+## Ways to use Aiven for Caching
+
+- Use Redis technology as a supplementary data store alongside a primary database
```
Comment (hoang-vo) [inline docs/products/caching/concepts/overview.md:48]: Here

File: docs/products/caching/howto/connect-java.md [hunk 10]
```diff
@@ -5,52 +5,52 @@ title: Connect with Java
...
 ## Variables
 
-These are the placeholders you will need to replace in the code sample:
+Replace the following placeholders in the code sample with actual values
+from your service overview page:
 
  | Variable    | Description                                                  |
  | ----------- | ------------------------------------------------------------ |
- | `REDIS_URI` | URL for the Redis connection, from the service overview page |
+ | `SERVICE_URI`| URI for the Redis connection, from the service overview page |
```
Comment (hoang-vo) [inline docs/products/caching/howto/connect-java.md:17]: AIven for Caching connection, Caching connection

File: docs/products/caching/howto/connect-java.md [hunk 11]
```diff
@@ -5,52 +5,52 @@ title: Connect with Java
...
-Central Repository](https://search.maven.org) and put them to `lib`
-folder
+If Maven is not installed, download the dependencies from the
+[Maven Central Repository](https://search.maven.org) and place them in the `lib`
+folder manually.
 
 ## Code
 
-Create a new file named `RedisExample.java`:
+Create a file named `RedisExample.java` and insert the code below,
```
Comment (hoang-vo) [inline docs/products/caching/howto/connect-java.md:37]: Maybe CachingExcample?

File: docs/products/caching/howto/connect-java.md [hunk 12]
```diff
@@ -5,52 +5,52 @@ title: Connect with Java
...
+This code creates a key named `key` with the value `hello world` without an expiration.
+It then retrieves this key from the caching service and outputs its value.
 
-Replace the placeholder with the **Redis URI** and compile and run the
+Replace the placeholder with the **CACHING_URI**, compile and run the
 code:
 
-```
+```bash
 javac -cp lib/*:. RedisExample.java && java -cp lib/*:. RedisExample REDIS_URI
```
Comment (hoang-vo) [inline docs/products/caching/howto/connect-java.md:49]: same here, the CACHING_URI and REDIS_URI probably need to change to SERVICE_URI?

**Review by hoang-vo — DISMISSED** [review]

File: docs/products/caching/howto/connect-java.md [hunk 13]
```diff
@@ -5,52 +5,52 @@ title: Connect with Java
...
 mvn org.apache.maven.plugins:maven-dependency-plugin:2.8:get -Dartifact=redis.clients:jedis:4.1.1:jar -Ddest=lib/jedis-4.1.1.jar \
 && mvn org.apache.maven.plugins:maven-dependency-plugin:2.8:get -Dartifact=org.apache.commons:commons-pool2:2.11.1:jar -Ddest=lib/commons-pool2-2.11.1.jar \
 && mvn org.apache.maven.plugins:maven-dependency-plugin:2.8:get -Dartifact=org.slf4j:slf4j-api:1.7.35:jar -Ddest=lib/slf4j-api-1.7.35.jar \
 && mvn org.apache.maven.plugins:maven-dependency-plugin:2.8:get -Dartifact=com.google.code.gson:gson:2.8.9:jar -Ddest=lib/gson-2.8.9.jar
 ```
 
-If there is no maven then download these dependencies from [Maven
-Central Repository](https://search.maven.org) and put them to `lib`
-folder
+If Maven is not installed, download the dependencies from the
```
Comment (nctbao) [inline docs/products/caching/howto/connect-java.md:31]: to be consistent with above
```suggestion
If `maven` is not installed, download the dependencies from the
```

File: docs/products/caching/howto/connect-java.md [hunk 14]
```diff
@@ -5,52 +5,52 @@ title: Connect with Java
...
+Create a file named `RedisExample.java` and insert the code below,
+substituting the placeholder with your Aiven for Caching URI:
 
 <CodeBlock language='java'>{MyComponentSource1}</CodeBlock>
 
-This code creates a key named `key` with the value `hello world` and no
-expiration time. Then, it gets the key back from Redis and prints its
-value.
+This code creates a key named `key` with the value `hello world` without an expiration.
+It then retrieves this key from the caching service and outputs its value.
```
Comment (nctbao) [inline docs/products/caching/howto/connect-java.md:43]: I wonder if we call it caching service or Caching service? Maybe we can get away with saying `from the service`

Comment (nctbao) [reply docs/products/caching/howto/connect-java.md:43]: for example in Dragonfly, we have it like this:
```
This code connects to Dragonfly, sets a key named key with the value hello world (without expiration), then retrieves and prints the value of this key.
```

File: docs/products/caching/concepts/restricted-redis-commands.md [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+title: Restricted Aiven for Caching commands
+---
+
+To ensure the stability and security of the caching environment, the Aiven for Caching service restricts certain commands.
+This section aims to provide you with a list of disabled/restricted commands.
+
+## Disabled commands
+
+The Aiven for Caching service has disabled the following commands, which pertain to
+the underlying Redis technology:
```
[the remark this answers is not in the document]
Comment (harshini-rangaswamy) [reply docs/products/caching/concepts/restricted-redis-commands.md:11]: Rephrased this.

File: docs/products/caching.md (hunk 1 above)
Comment (harshini-rangaswamy) [reply docs/products/caching.md:5]: Done

File: docs/products/caching/concepts/lua-scripts-caching.md (hunk 2 above)
Comment (harshini-rangaswamy) [reply docs/products/caching/concepts/lua-scripts-caching.md:5]: Done

File: docs/products/caching/concepts/memory-usage.md (hunk 4 above)
Comment (harshini-rangaswamy) [reply docs/products/caching/concepts/memory-usage.md:5]: Added this - This service is compatible with legacy Redis® OSS up to version 7.2.4, facilitating seamless transitions and compatibility.


is that what you intended?

File: docs/products/caching/concepts/overview.md (hunk 6 above)
Comment (harshini-rangaswamy) [reply docs/products/caching/concepts/overview.md:5]: Addded this. - This service is compatible with legacy Redis® OSS up to version 7.2.4, facilitating seamless transitions and compatibility.

File: docs/products/caching/concepts/overview.md (hunk 7 above)
Comment (harshini-rangaswamy) [reply docs/products/caching/concepts/overview.md:23]: done.

File: docs/products/caching/howto/connect-java.md (hunk 14 above)
Comment (harshini-rangaswamy) [reply docs/products/caching/howto/connect-java.md:43]: Good point. I rephrased similar to the suggested line.

**Review by ArthurFlag — APPROVED** [review]

**Review by ArthurFlag — DISMISSED** [review]
I still see occurrences of Redis in 50 files, is this expected?

**Review by wojcik-dorota — APPROVED** [review]
LGTM

**Pull request merged**